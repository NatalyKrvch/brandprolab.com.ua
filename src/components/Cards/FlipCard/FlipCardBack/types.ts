export interface FlipCardBackProps {
  header: string;
  list: string[];
}
