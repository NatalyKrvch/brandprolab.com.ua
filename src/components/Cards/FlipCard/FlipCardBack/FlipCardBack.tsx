import clsx from 'clsx';
import { twMerge } from 'tailwind-merge';

import {
  FLIP_CARD_BACK_HEADER_TEST_ID,
  FLIP_CARD_BACK_LIST_ITEM_TEST_ID,
  FLIP_CARD_BACK_LIST_TEST_ID,
} from '@/lib/testIDs';

import { SimpleCard } from '../../SimpleCard';
import { FlipCardBackProps } from './types';

const FlipCardBack = ({ header, list }: FlipCardBackProps) => {
  return (
    <SimpleCard
      className={twMerge(
        clsx('flex h-full flex-col gap-6 p-10', 'bg-blue-dark'),
      )}
    >
      <p
        className={twMerge(
          clsx(
            'text-lg font-bold leading-5 text-teal-medium',
            'tablet:text-2xl tablet:leading-6',
            'desktop:text-28 desktop:leading-7',
          ),
        )}
        data-testid={FLIP_CARD_BACK_HEADER_TEST_ID}
      >
        {header}
      </p>
      <ul
        className={twMerge(
          clsx(
            'list-disc pl-5',
            'text-sm font-normal leading-18 text-white',
            'tablet:text-base tablet:leading-6',
            'desktop:text-lg desktop:leading-5',
          ),
        )}
        data-testid={FLIP_CARD_BACK_LIST_TEST_ID}
      >
        {list.map(item => (
          <li
            className="my-2"
            key={item}
            data-testid={FLIP_CARD_BACK_LIST_ITEM_TEST_ID}
          >
            {item}
          </li>
        ))}
      </ul>
    </SimpleCard>
  );
};

export default FlipCardBack;
