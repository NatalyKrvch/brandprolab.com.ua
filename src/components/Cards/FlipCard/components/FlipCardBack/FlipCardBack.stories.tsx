import type { Meta, StoryObj } from '@storybook/react';

import FlipCardBack from './FlipCardBack';

const meta: Meta<typeof FlipCardBack> = {
  title: 'Components/FlipCardBack',
  component: FlipCardBack,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    header: {
      control: 'text',
      description: 'The title displayed at the top of the card.',
      table: {
        type: { summary: 'string' },
      },
    },
    list: {
      control: { type: 'object' },
      description: 'A list of items displayed in the card.',
      table: {
        type: { summary: 'string[]' },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof FlipCardBack>;

export const Default: Story = {
  args: {
    header: 'Основні переваги',
    list: [
      'Глибокий аналіз',
      'Професійний підхід',
      'Перевірена інформація',
      'Експертні висновки',
    ],
  },
};
