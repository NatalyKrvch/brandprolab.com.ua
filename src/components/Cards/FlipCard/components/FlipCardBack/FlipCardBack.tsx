import { SimpleCard } from '@/components/Cards/SimpleCard';
import {
  FLIP_CARD_BACK_HEADER_TEST_ID,
  FLIP_CARD_BACK_LIST_ITEM_TEST_ID,
  FLIP_CARD_BACK_LIST_TEST_ID,
} from '@/lib/testIDs';

import type { FlipCardBackProps } from './types';

const FlipCardBack = ({ header, list }: FlipCardBackProps) => {
  return (
    <SimpleCard className="flex h-full flex-col gap-6 bg-blue-dark p-10">
      <p
        className="text-lg font-bold leading-5 text-teal-medium tablet:text-2xl tablet:leading-6 desktop:text-28 desktop:leading-7"
        data-testid={FLIP_CARD_BACK_HEADER_TEST_ID}
      >
        {header}
      </p>
      <ul
        className="list-disc pl-5 text-sm font-normal leading-18 text-white tablet:text-base tablet:leading-6 desktop:text-lg desktop:leading-5"
        data-testid={FLIP_CARD_BACK_LIST_TEST_ID}
      >
        {list.map(item => (
          <li
            className="my-2"
            key={item}
            data-testid={FLIP_CARD_BACK_LIST_ITEM_TEST_ID}
          >
            {item}
          </li>
        ))}
      </ul>
    </SimpleCard>
  );
};

export default FlipCardBack;
