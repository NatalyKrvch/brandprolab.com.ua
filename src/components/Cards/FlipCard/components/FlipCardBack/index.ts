export { default as FlipCardBack } from './FlipCardBack';
