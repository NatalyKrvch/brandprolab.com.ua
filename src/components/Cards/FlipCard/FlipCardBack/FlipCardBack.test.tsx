import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import {
  FLIP_CARD_BACK_HEADER_TEST_ID,
  FLIP_CARD_BACK_LIST_ITEM_TEST_ID,
  FLIP_CARD_BACK_LIST_TEST_ID,
  SIMPLE_CARD_TEST_ID,
} from '@/lib/testIDs';

import FlipCardBack from './FlipCardBack';

describe('FlipCardBack component', () => {
  const defaultProps = {
    header: 'Test Header',
    list: ['Item 1', 'Item 2', 'Item 3'],
  };

  it('renders the FlipCardBack component', () => {
    const { asFragment } = render(<FlipCardBack {...defaultProps} />);
    expect(screen.getByTestId(SIMPLE_CARD_TEST_ID)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the header correctly', () => {
    const { asFragment } = render(<FlipCardBack {...defaultProps} />);
    expect(screen.getByTestId(FLIP_CARD_BACK_HEADER_TEST_ID)).toHaveTextContent(
      'Test Header',
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the list correctly', () => {
    const { asFragment } = render(<FlipCardBack {...defaultProps} />);
    expect(screen.getByTestId(FLIP_CARD_BACK_LIST_TEST_ID)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct number of list items', () => {
    const { asFragment } = render(<FlipCardBack {...defaultProps} />);
    const listItems = screen.getAllByTestId(FLIP_CARD_BACK_LIST_ITEM_TEST_ID);

    expect(listItems).toHaveLength(defaultProps.list.length);
    listItems.forEach((item, index) => {
      expect(item).toHaveTextContent(defaultProps.list[index]);
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
