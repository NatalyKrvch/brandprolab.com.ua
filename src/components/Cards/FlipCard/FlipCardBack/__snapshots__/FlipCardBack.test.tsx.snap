// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlipCardBack component renders the FlipCardBack component 1`] = `
<DocumentFragment>
  <div
    class="flex h-full flex-col gap-6 p-10 bg-blue-dark rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p
      class="text-lg font-bold leading-5 text-teal-medium tablet:text-2xl tablet:leading-6 desktop:text-28 desktop:leading-7"
      data-testid="flip-card-back-header-test-id"
    >
      Test Header
    </p>
    <ul
      class="list-disc pl-5 text-sm font-normal text-white tablet:text-base tablet:leading-6 desktop:text-lg desktop:leading-5"
      data-testid="flip-card-back-list-test-id"
    >
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`FlipCardBack component renders the correct number of list items 1`] = `
<DocumentFragment>
  <div
    class="flex h-full flex-col gap-6 p-10 bg-blue-dark rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p
      class="text-lg font-bold leading-5 text-teal-medium tablet:text-2xl tablet:leading-6 desktop:text-28 desktop:leading-7"
      data-testid="flip-card-back-header-test-id"
    >
      Test Header
    </p>
    <ul
      class="list-disc pl-5 text-sm font-normal text-white tablet:text-base tablet:leading-6 desktop:text-lg desktop:leading-5"
      data-testid="flip-card-back-list-test-id"
    >
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`FlipCardBack component renders the header correctly 1`] = `
<DocumentFragment>
  <div
    class="flex h-full flex-col gap-6 p-10 bg-blue-dark rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p
      class="text-lg font-bold leading-5 text-teal-medium tablet:text-2xl tablet:leading-6 desktop:text-28 desktop:leading-7"
      data-testid="flip-card-back-header-test-id"
    >
      Test Header
    </p>
    <ul
      class="list-disc pl-5 text-sm font-normal text-white tablet:text-base tablet:leading-6 desktop:text-lg desktop:leading-5"
      data-testid="flip-card-back-list-test-id"
    >
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`FlipCardBack component renders the list correctly 1`] = `
<DocumentFragment>
  <div
    class="flex h-full flex-col gap-6 p-10 bg-blue-dark rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p
      class="text-lg font-bold leading-5 text-teal-medium tablet:text-2xl tablet:leading-6 desktop:text-28 desktop:leading-7"
      data-testid="flip-card-back-header-test-id"
    >
      Test Header
    </p>
    <ul
      class="list-disc pl-5 text-sm font-normal text-white tablet:text-base tablet:leading-6 desktop:text-lg desktop:leading-5"
      data-testid="flip-card-back-list-test-id"
    >
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        class="my-2"
        data-testid="flip-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
